### PR TITLE
feat(vsc): add tree view telemetery in vsc

### PR DESCRIPTION
**Pull request by 1openwindow**
work item: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/9888182

### CODE REVIEW

**Review by HuihuiWu-Microsoft — APPROVED** [review]

File: packages/vscode-extension/src/handlers.ts [hunk 1]
```diff
@@ -759,11 +693,11 @@ export async function cmdHdlLoadTreeView(context: ExtensionContext) {
     try {
       switch (node.contextValue) {
         case "signedinM365": {
-          signOutM365();
+          signOutM365(true);
```
Comment (chagong) [inline packages/vscode-extension/src/handlers.ts:696]: we do sign in but the function name here is "signOutXXX()", do we do it on purpose?

File: packages/vscode-extension/src/telemetry/extTelemetryEvents.ts [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
+  QuickStart = "quick-start",
+
+  Samples = "samples",
+
+  Documentation = "documentation",
+
   LoginStart = "login-start",
   Login = "login",
 
+  LoginAzure = "login-azure",
```
Comment (chagong) [inline packages/vscode-extension/src/telemetry/extTelemetryEvents.ts:13]: Let's use "login-start" and "login" for event name, and add "account-type" property to indicate account type.

File: packages/fx-core/src/core/index.ts [hunk 3]
```diff
@@ -486,7 +487,8 @@ class CoreImpl implements Core {
                 return ok(null);
             };
 
-            const signinM365Callback = async (): Promise<Result<null, FxError>> => {
+            const signinM365Callback = async (args?: any[]): Promise<Result<null, FxError>> => {
+                this.ctx?.telemetryReporter?.sendTelemetryEvent(TelemetryEvent.LoginM365, getTriggerFromProperty(args));
```
Comment (chagong) [inline packages/fx-core/src/core/index.ts:491]: And we should have the following properties according to PM's spec: 
"component", "hashed-userid", "internal": "internal-alias"

Comment (1openwindow) [reply packages/fx-core/src/core/index.ts:491]: @chagong can you share the spec? I try to understand what these properties are.

**Review by jayzhang — APPROVED** [review]

**Pull request merged**